### PR TITLE
Introduce new config variable to control whether DAG processor outputs to stdout

### LINKED ISSUE #37122: Allow Dag Processor Manager to log to stdout
### Description

On Airflow 2.8.1, the DAG processor manager no longer logs out to stdout (whereas it did on 2.6.2).

### Use case/motivation

We currently have a configuration option for the DAG processor to log to stdout via `AIRFLOW__LOGGING__DAG_PROCESSOR_LOG_TARGET`. A similar configuration `AIRFLOW__LOGGING__DAG_PROCESSOR_MANAGER_LOG_TARGET` could achieve that.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Owen-CH-Leung**
fixes #37122 

This PR adds a new config variable under `logging` called `dag_processor_manager_log_stdout`, which is set to `False` by default. If being set to `True`, the `DagFileProcessorAgent` will also output logs to the stdout of `airflow scheduler`, in addition to writing logs to the designated log file defined by `dag_processor_manager_log_location`

### CODE REVIEW

File: tests/dag_processing/test_job_runner.py [hunk 1]
```diff
@@ -1460,3 +1460,20 @@ def test_launch_process(self):
         processor_agent._process.join()
 
         assert os.path.isfile(log_file_loc)
+
+    @conf_vars({("logging", "dag_processor_manager_log_stdout"): "True"})
```
Comment (dirrao) [inline tests/dag_processing/test_job_runner.py:1636]: can you add the test case for when it is disabled?

Comment (Owen-CH-Leung) [reply tests/dag_processing/test_job_runner.py:1636]: Sure. Added

File: airflow/config_templates/config.yml [hunk 2]
```diff
@@ -857,6 +857,13 @@ logging:
       type: string
       example: ~
       default: "{AIRFLOW_HOME}/logs/dag_processor_manager/dag_processor_manager.log"
+    dag_processor_manager_log_stdout:
+      description: |
+        Whether DAG processor manager will write logs to stdout
+      version_added: 2.8.1
```
Comment (eladkal) [inline airflow/config_templates/config.yml]:
```suggestion
      version_added: 2.9.0
```

Comment (Owen-CH-Leung) [reply airflow/config_templates/config.yml]: Thanks. Changed to `2.9.0`

**Comment by potiuk** [comment]
CAn you resolve conflicts and rebase please?

**Comment by Owen-CH-Leung** [comment]
> CAn you resolve conflicts and rebase please?

Sure & rebased

**Review by potiuk — APPROVED** [review]

**Comment by dimon222** [comment]
How to select output ONLY to stdout? If I write to both locations I risk getting OOM in container environment and I don't want to persist it perpetually in PVC.

**Pull request merged**